File: app/src/main/java/com/mifos/apache/fineract/data/models/deposit/Term.java
package com.mifos.apache.fineract.data.models.deposit;


public class Term {

    private Integer period;
    private TimeUnit timeUnit;
    private InterestPayable interestPayable;

    public Term() {
        super();
    }

    public Integer getPeriod() {
        return this.period;
    }

    public void setPeriod(final Integer period) {
        this.period = period;
    }

    public String getTimeUnit() {
        if (this.timeUnit != null) {
            return this.timeUnit.name();
        } else {
            return null;
        }
    }

    public void setTimeUnit(final String timeUnit) {
        if (timeUnit != null) {
            this.timeUnit = TimeUnit.valueOf(timeUnit);
        }
    }

    public String getInterestPayable() {
        return this.interestPayable.name();
    }

    public void setInterestPayable(final String interestPayable) {
        this.interestPayable = InterestPayable.valueOf(interestPayable);
    }
}

File: app/src/main/java/org/apache/fineract/data/models/deposit/Term.kt
package org.apache.fineract.data.models.deposit

import com.google.gson.annotations.SerializedName


data class Term(
    @SerializedName("period") var period: Int? = null
) {

    private var timeUnit: TimeUnit? = null
    private var interestPayable: InterestPayable? = null

    fun getTimeUnit(): String? {
        return if (this.timeUnit != null) {
            this.timeUnit!!.name
        } else {
            null
        }
    }

    fun setTimeUnit(timeUnit: String?) {
        if (timeUnit != null) {
            this.timeUnit = TimeUnit.valueOf(timeUnit)
        }
    }

    fun getInterestPayable(): String {
        return this.interestPayable!!.name
    }

    fun setInterestPayable(interestPayable: String) {
        this.interestPayable = InterestPayable.valueOf(interestPayable)
    }
}